core/src/main/java/hudson/model/MyViewsProperty.java

```diff
@@ -71,12 +78,13 @@
     private transient ViewGroupMixIn viewGroupMixIn;
 
     @DataBoundConstructor
-    public MyViewsProperty(String primaryViewName) {
+    public MyViewsProperty(@CheckForNull String primaryViewName) {
         this.primaryViewName = primaryViewName;
+        readResolve(); // initialize fields
```

Comment:
Do you have any stack traces or reports from users that happened because `readResolve` wasn't called? I don't know how much this constructor is used but it seems surprising that there isn't already an issue for it.

Reply:
I have not found usages of this constructor in API. In Jenkins Core view properties work somehow though I do not see explicit reload() logic in the data submission

Reply:
But in tests it has failed with flying colors